### PR TITLE
Add event lines to plots

### PR DESCRIPTION
**Pull request by amitschang**
Adds lines to plotter for events.

**Notes:**
* Unfortunately since the events most likely don't come from hardware, we need to send a separate request to get them (the sensor request specifies name). Only alternatives I think would be to either treat events separately in the history endpoint (not sure I like that) or just retrieve all data.
* I noticed the shape of data might not be ideal, iso `{'name': [{...}], 'name2': [{...}]}`, might want to change backend to `[{'name': 'name', ...}, {'name': 'name2', ...}]`. This would replace the awkwardness of the `flatmap` currently done on events, and also preserve the name in each record. *can do in separate change with backend*
* The time axis was categorical, so lines not exactly corresponding to a point would not render. It has changed to numeric and with a formatter to render as time.
* I couldn't figure out how to make labels vertical or better-yet show up as a tooltip, so any useful log message ends up cluttering and possibly unreadable...

Example SS:

image

### CODE REVIEW

File: app/routes/devices.$id.hardware.$hardware_name.history.tsx [hunk 1]
```diff
@@ -51,17 +52,27 @@ export async function loader({ params, request }: LoaderFunctionArgs) {
...
+    },
+    body: {
       vials,
       properties,
+      kinds: ["sensor"],
+    },
+    client: evolverClient,
+  });
+
+  const events = await Evolver.history({
```
Comment (imaitland) [inline app/routes/devices.$id.hardware.$hardware_name.history.tsx:64]: You can use Promise.all to parralleize these two request.
```
const [historyData, eventsData] = Promise.allSettled([await Evolver.history({whatever}), await Evolver.history({whatever + kinds["event"]})])
```

File: app/components/LineChart.tsx [hunk 2]
```diff
@@ -57,6 +81,14 @@ export const HardwareLineChart = ({ vials, rawData, property = "raw" }) => {
               connectNulls={true}
             />
           ))}
+          {eventData.map((event) => (
+            <ReferenceLine
+              key={event}
+              x={event.timestamp}
+              label={event.data.message}
```
Comment (imaitland) [inline app/components/LineChart.tsx:88]: Label will take a component as prop, and daisyUI provides a tooltip: https://daisyui.com/components/tooltip/,

```
label = {() => {
  return (
    <div className="tooltip" data-tip={event.data.message}>
      {event.data.message.length > 20 ? `${event.data.message.slice(0, 20)}...` : event.data.message}
    </div>
  )
}}
```

File: app/routes/devices.$id.hardware.$hardware_name.history.tsx [hunk 3]
```diff
@@ -82,6 +93,11 @@ export default function Hardware() {
     hardwareHistory[0].data,
   ).filter((property) => excludedProperties.includes(property) === false);
 
+  // shape of data is not ideal here, we have a struct mapping event name to
+  // array of events. Here we drop name, probably change to backend could keep
+  // the name in the struct
+  const allEvents = flatMap(events);
+
```
Comment (imaitland) [inline app/routes/devices.$id.hardware.$hardware_name.history.tsx:100]: Interesting, did you test this change with a chart that shows multiple vials' histories?

File: app/components/LineChart.tsx [hunk 4]
```diff
@@ -57,6 +81,14 @@ export const HardwareLineChart = ({ vials, rawData, property = "raw" }) => {
               connectNulls={true}
             />
           ))}
+          {eventData.map((event) => (
+            <ReferenceLine
```
Comment (imaitland) [inline app/components/LineChart.tsx:85]: Nice!

File: app/components/LineChart.tsx (hunk 2 above)
Comment (amitschang) [reply app/components/LineChart.tsx:88]: Looks cool, but when I do this I don't see any text at all. I tried with plain div and span elements like this, doesn't seem to show. There may be some requirements it has on the underlying element...

**Comment by amitschang** [comment]
Made updates and re-requesting review. To do the label thing well looks to require more effort, so I think what we can do is merge this in (if otherwise good) and make label change as an enhancement (it affects only a single line in this PR)

**Review by imaitland — APPROVED** [review]
LGTM

**Comment by amitschang** [comment]
thanks!

**Pull request merged**